nearswap/src/internal.rs

```diff
@@ -76,102 +59,122 @@ impl NearSwap {
             "Swapping_in {} {} -> {} ynear -> {} {}",
             tokens_in, t_in, near_swap, tokens2_out, t_out
         );
-        return (p_in, p_out, near_swap, tokens2_out);
+        return tokens2_out;
     }
 
-    pub(crate) fn _price_swap_tokens_out(
-        &self,
-        t_in: &AccountId,
-        t_out: &AccountId,
-        tokens_out: Balance,
-    ) -> (Pool, Pool, Balance, Balance) {
-        assert!(tokens_out > 0, "E2: balance arguments must be >0");
-        assert_ne!(t_in, t_out, "E9: can't swap same tokens");
-        let p_in = self.get_pool(&t_in);
-        let p_out = self.get_pool(&t_out);
-        let near_swap = self.calc_in_amount(tokens_out, p_out.ynear, p_out.tokens);
-        let tokens1_to_pay = self.calc_in_amount(near_swap, p_in.tokens, p_in.ynear);
-        println!(
-            "Swapping_out {} {} -> {} ynear -> {} {}",
-            tokens1_to_pay, t_in, near_swap, tokens_out, t_out
-        );
-        return (p_in, p_out, near_swap, tokens1_to_pay);
+    /// returns swap out amount and fee.
+    pub(crate) fn calc_out_with_fee(&self, mut x: u128,  X: u128, Y: u128) -> (u128, u128) {
```

Comment:
let's move this function up, just below `calc_out_amount`

Reply:
ok